### PR TITLE
[Merged by Bors] - feat: port Topology.Constructions

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Constructions.lean [hunk 1]
```diff
@@ -1459,22 +1345,21 @@ the `f i` as `i : ι` varies is simply the topology on `X` induced by `g : X →
...
   congr 1
   funext
-  erw [induced_compose]
+  erw [induced_compose]; rfl
 #align inducing_infi_to_pi inducing_infᵢ_to_pi
 
 variable [Finite ι] [∀ i, DiscreteTopology (π i)]
 
 /-- A finite product of discrete spaces is discrete. -/
 instance PiCat.discreteTopology : DiscreteTopology (∀ i, π i) :=
```
Comment (urkud) [inline Mathlib/Topology/Constructions.lean:1359]: I don't know why `mathport` translates `Pi.` to `PiCat.`. Should I revert this?

Comment (jcommelin) [reply Mathlib/Topology/Constructions.lean:1359]: Yes, please. There is a PR to mathport that tries to fix this.

Comment (jcommelin) [reply Mathlib/Topology/Constructions.lean:1359]: I merged that PR right now. So in a couple of hours, this problem shouldn't occur anymore.

Comment (urkud) [reply Mathlib/Topology/Constructions.lean]: AFAIR, full cycle takes ~5h.

File: Mathlib/Topology/Constructions.lean [hunk 2]
```diff
@@ -0,0 +1,1525 @@
...
+    (TopologicalSpace.induced Prod.snd _)).trans <|
+    continuous_induced_rng.and continuous_induced_rng
+#align continuous_prod_mk continuous_prod_mk
+
+-- porting note: todo: restore @[continuity]
+theorem continuous_fst : Continuous (@Prod.fst α β) :=
+  (continuous_prod_mk.1 continuous_id).1
+#align continuous_fst continuous_fst
+
+/-- Postcomposing `f` with `prod.fst` is continuous -/
```
Comment (jcommelin) [inline Mathlib/Topology/Constructions.lean:332]:
```suggestion
/-- Postcomposing `f` with `Prod.fst` is continuous -/
```
and similar changes below.

**Review by jcommelin — COMMENTED** [review]
bors d+

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**